### PR TITLE
Add docker dev prod workflow guide

### PR DESCRIPTION
**Pull request by carolineechen**

### CODE REVIEW

**Comment by carolineechen** [comment]
* **#1094** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1094?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @carolineechen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/guides/docker-dev.rst [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+Afterwards, we provide a script that shows how to easily set up and
+toggle between these two settings, using the same cluster setup.
+
+In this example, we are going to be using the `DJLServing 0.27.0 with
+DeepSpeed
+0.12.6 <https://github.com/aws/deep-learning-containers/blob/master/available_images.md#large-model-inference-containers>`__
+Container, which includes HuggingFace Tranformers (4.39.0), Diffusers
+(0.16.0), and Accelerate (0.28.0). We will use both the container
+version of these packages, as well as local editable versions, of the
+diffusers and transformers packages to showcase both production ready
```
Comment (dongreenberg) [inline docs/guides/docker-dev.rst:26]: Delete "of the diffusers and transformers packages"

File: docs/guides/docker-dev.rst [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+    cluster = rh.ondemand_cluster(
+        name="diffusers_docker",
+        image_id="docker:djl-inference:0.27.0-deepspeed0.12.6-cu121",
+        instance_type="g5.8xlarge",
+        provider="aws",
+    )
+    cluster.up_if_not()
+
+The function we’ll be using in our demo is ``is_transformers_available``
+from ``diffusers.utils``. We’ll first show what using this function like
```
Comment (dongreenberg) [inline docs/guides/docker-dev.rst:72]:
```suggestion
from ``diffusers.utils``. We’ll first show what using this function
```

File: docs/guides/docker-dev.rst [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+directly on the box (e.g. a production setting) looks like. After, we’ll
+show the case if we had a local version of the repositories, that we
+modified locally, and wanted to test out the change on the cluster.
```
Comment (dongreenberg) [inline docs/guides/docker-dev.rst:75]:
```suggestion
show the case if we had local versions of the repositories that we'd
modified and wanted to test out our changes on the cluster.
```

File: docs/guides/docker-dev.rst [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+
+    from diffusers.utils import is_transformers_available
+
+Production Workflow
+-------------------
+
+The core of the production workflow is that the Docker image consists of
+the exact packages and versions that you want. Regardless of whether or
+not one has local edits, these edits should **not** be used in the
+production workflow.
```
Comment (dongreenberg) [inline docs/guides/docker-dev.rst:87]:
```suggestion
The core of the production workflow is that the Docker image already contains
the exact packages and versions we want, probably published into the registry in CI/CD.
We don't want to perform any installs or code changes within the image throughout
execution so we preserve exact reprodicubility.
```

File: docs/guides/docker-dev.rst [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+
+**NOTE**: By default, Ray and Runhouse are installed on the ondemand
+cluster during setup time. This may be the most updated version of
+Runhouse, or the version you have cloned and installed locally. To make
+sure that your version is pinned for production use cases, please make
+sure that you have Runhouse and Ray included on the Docker container;
+the setup will skip the installations if it detects that they are
+already installed on the cluster.
```
Comment (dongreenberg) [inline docs/guides/docker-dev.rst:95]:
```suggestion
**NOTE**: By default, Ray and Runhouse are installed on the ondemand
cluster during setup time (generally attempting to match the versions you 
have locally), unless we detect that they're already present. To make
sure that no installs occur in production, please make
sure that you have Runhouse and Ray installed in your docker image.
```

**Review by dongreenberg — APPROVED** [review]

**Pull request merged**